tests/publiccloud/img_proof.pm

```diff
@@ -15,35 +15,35 @@ use Mojo::Base 'publiccloud::basetest';
 use testapi;
 use Mojo::File 'path';
 use Mojo::JSON;
-use publiccloud::utils "select_host_console";
+use publiccloud::utils qw(select_host_console is_ondemand);
 
 # for not released versions we need to exclude :
 # * test_sles_kernel_version - test checking CONFIG_SUSE_PATCHLEVEL correctness but during chat with kernel devs it was clarified that they don't care about this variable till release
 # * test_sles_multipath_off - TODO : simply not exists in current version of publiccloud_tools image used
-our $test_sles_for_dev = ',test_soft_reboot,test_sles_license,test_sles_root_pass,test_hard_reboot,test_sles_hostname,test_sles_haveged,test_sles_lscpu,test_sles_motd';
+our $test_sles_for_dev = 'test_soft_reboot,test_sles_license,test_sles_root_pass,test_hard_reboot,test_sles_hostname,test_sles_haveged,test_sles_lscpu,test_sles_motd';
 # for not released versions we need to exclude :
 # * test_sles_repos - not released version repo names have initially names omit 'Beta/Snapshot' titles. This test trying
 # to compare repo name with VERSION which has 'Beta/Snapshot' so test will always fail
-our $test_sles_on_demand_for_dev = ',test_sles_wait_on_registration,test_refresh,test_sles_smt_reg,test_sles_guestregister';
+our $test_sles_on_demand_for_dev = 'test_sles_wait_on_registration,test_refresh,test_sles_smt_reg,test_sles_guestregister';
 
 our $azure_byos_updates      = 'test_sles,test_sles_azure';
-our $azure_on_demand_updates = 'test_sles_wait_on_registration,test_sles,test_sles_on_demand,test_sles_azure';
+our $azure_on_demand_updates = 'test_sles,test_sles_on_demand,test_sles_azure';
 
-our $azure_byos      = 'test_sles_azure' . $test_sles_for_dev;
-our $azure_on_demand = 'test_sles_wait_on_registration,test_sles_azure' . $test_sles_on_demand_for_dev;
+our $azure_byos      = $test_sles_for_dev . ',test_sles_azure';
+our $azure_on_demand = $test_sles_for_dev . ',' . $test_sles_on_demand_for_dev . ',test_sles_azure';
```

Comment:
Can we create a sub to wrap the format here.
something like 
```
sub formatter {
my @tests = @_;
return join ",", @tests;
}
our $azure_on_demand = formatter $test_sles_for_dev,$test_sles_on_demand_for_dev,test_sles_azure;
```
wdyt?

Reply:
sorry if I misunderstood you but for it sounds like you trying to suggest something what we already has https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/provider.pm#L187